src/Core/Manifest/ModuleResource.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace SilverStripe\Core\Manifest;
+
+use InvalidArgumentException;
+use SilverStripe\Core\Injector\Injector;
+
+/**
+ * Experimental API; This may change at any time
+ */
+class ModuleResource
+{
+    /**
+     * @var Module
+     */
+    protected $module = null;
+
+    /**
+     * Path to this resource relative to the module (no leading slash)
+     *
+     * @var string
+     */
+    protected $relativePath = null;
+
+    /**
+     * ModuleResource constructor.
+     *
+     * @param Module $module
+     * @param string $relativePath
+     */
+    public function __construct(Module $module, $relativePath)
+    {
+        $this->module = $module;
+        $this->relativePath = ltrim($relativePath, Module::TRIM_CHARS);
+        if (empty($this->relativePath)) {
+            throw new InvalidArgumentException("Resource cannot have empty path");
+        }
+    }
+
+    /**
+     * @return string
+     */
+    public function getPath()
```

Comment:
Note that for path operations, it'll get you the *original* file, not the copied one.

Reply:
Yep, I'll document this in the code a bit more concisely.